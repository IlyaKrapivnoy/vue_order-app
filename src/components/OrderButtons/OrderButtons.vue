<template>
  <div class="order">
    <button
      v-for="(term, index) in orderTerms"
      :key="index"
      @click="handleClick(term)"
      :class="{ active: term === selectedTerm }"
    >
      Order by {{ term }}
    </button>
  </div>
</template>

<script setup lang="ts">
  import { defineEmits, ref } from 'vue';
  import OrderTerm from '@/types/OderTerm';

  const orderTerms: OrderTerm[] = ['location', 'title', 'salary'];
  const selectedTerm = ref<OrderTerm>('title');

  const emit = defineEmits(['onOrderClick']);

  const handleClick = (term: OrderTerm) => {
    selectedTerm.value = term;
    emit('onOrderClick', term);
  };
</script>

<style scoped>
  .order {
    margin-top: 20px;
  }

  .active {
    background-color: #1195c9;
    color: white;
  }
</style>
