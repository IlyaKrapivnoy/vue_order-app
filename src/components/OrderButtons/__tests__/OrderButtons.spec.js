import { mount } from '@vue/test-utils';
import OrderButtons from '@/components/OrderButtons/OrderButtons.vue';

describe('OrderButtons.vue', () => {
  it('emits the "onOrderClick" event when a button is clicked', async () => {
    const wrapper = mount(OrderButtons);

    // Simulate a click on the first button
    await wrapper.findAll('button').at(0).trigger('click');

    // Assert that the "onOrderClick" event was emitted with the expected argument
    expect(wrapper.emitted('onOrderClick')).toBeTruthy();
    expect(wrapper.emitted('onOrderClick')[0]).toEqual(['location']);
  });

  it('emits the "onOrderClick" event with "location" when the "Order by location" button is clicked', async () => {
    const wrapper = mount(OrderButtons);
    await wrapper.findAll('button').at(0).trigger('click');
    expect(wrapper.emitted('onOrderClick')).toBeTruthy();
    expect(wrapper.emitted('onOrderClick')[0]).toEqual(['location']);
  });

  it('emits the "onOrderClick" event with "title" when the "Order by title" button is clicked', async () => {
    const wrapper = mount(OrderButtons);
    await wrapper.findAll('button').at(1).trigger('click');
    expect(wrapper.emitted('onOrderClick')).toBeTruthy();
    expect(wrapper.emitted('onOrderClick')[0]).toEqual(['title']);
  });

  it('emits the "onOrderClick" event with "salary" when the "Order by salary" button is clicked', async () => {
    const wrapper = mount(OrderButtons);
    await wrapper.findAll('button').at(2).trigger('click');
    expect(wrapper.emitted('onOrderClick')).toBeTruthy();
    expect(wrapper.emitted('onOrderClick')[0]).toEqual(['salary']);
  });
});
