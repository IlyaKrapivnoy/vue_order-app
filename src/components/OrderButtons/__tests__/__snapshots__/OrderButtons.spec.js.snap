// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderButtons.vue matches the snapshot 1`] = `<div class="order"><button class=""> Order by location</button><button class="active"> Order by title</button><button class=""> Order by salary</button></div>`;
